code/ch3.3/src/cpu.rs

```diff
@@ -229,7 +229,7 @@ impl CPU {
 
     pub fn load_and_run(&mut self, program: Vec<u8>) {
         self.load(program);
-        self.reset();
+        self.program_counter = self.mem_read_u16(0xFFFC);
```

Comment:
you don't have to hack load_and_run to make test past, you can use load, reset and run separately in the test case